### PR TITLE
[XM] Add NSValidatedUserInterfaceItem, NSCloudSharingValidation, NSUs…

### PR DESCRIPTION
**Pull request by chamons**
…erInterfaceValidations, and NSMenuValidation

- https://bugzilla.xamarin.com/show_bug.cgi?id=18359

### CODE REVIEW

**Comment by spouliot** [comment]
small nitpick but it would be nicer (for git history) not to move the commented code before re-enabling it. Otherwise it LGTM :)

**Comment by chamons** [comment]
I originally had it all behind a XAMCORE_2_0 but removed that and forgot to move back.

**Comment by chamons** [comment]
I'll fix shortly.

**Comment by chamons** [comment]
Once we squash and merge git history should be safe now.

**Comment by chamons** [comment]
build

File: src/appkit.cs [hunk 1]
```diff
@@ -23683,4 +23687,20 @@ interface NSToolTipOwner
...
+		bool ValidateUserInterfaceItem (INSValidatedUserInterfaceItem item);
+	}
+
+	[Protocol (IsInformal=true)]
+	interface NSMenuValidation
+	{
+		[Abstract]
+		[Export ("validateUserInterfaceItem:")]
+		bool ValidateUserInterfaceItem (INSValidatedUserInterfaceItem item);
+	}
```
Comment (spouliot) [inline src/appkit.cs]: do we need both ? again it's not an API I ever used but the bug report mention one of them is needed - and it turns out they are identical (but in name) to implement.

Is there a case where the informal must be used that the _normal_ protocol would not cover ?

Comment (chamons) [inline src/appkit.cs]: NSMenuValidation is just totally wrong. Copy and paste error. Good catch.

Comment (chamons) [inline src/appkit.cs]: Fixed. Sorry about that.

**Pull request merged**